pkg/kubelet/volumes.go

```diff
@@ -77,12 +77,18 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.BoundPod) (volumeMap, error) {
 	for i := range pod.Spec.Volumes {
 		volSpec := &pod.Spec.Volumes[i]
 
+		podRef, err := api.GetReference(pod)
```

Comment:
IMO it should be possible to take references to all pods kubelet knows about, including ones from config file, http, etc., so those methods should test that this is possible. This is because we'll need to checkpoint & recover from crashes, even if the config file changes while kubelet restarts. But that should be an extremely rare edge case, so I'm not sure if it's worth changing it now.